src/app/devices/deviceContent/components/digitalTwin/digitalTwinInterfaces.spec.tsx

```diff
@@ -44,12 +76,107 @@ describe('login/components/connectivityPane', () => {
+                isDigitalTwinLoading: false,
+                modelDefinitionWithSource: {
+                    isModelValid: false,
+                    modelDefinition: null,
+                    source: REPOSITORY_LOCATION_TYPE.Local,
+                },
+                modelId: 'dtmi:__azureiot:sampleModel;1'
+            }),
+            true,
+            true).find(DigitalTwinInterfaces);
+
+        const h4 = digitalTwinInterfaces.find('h4');
+        expect(h4).toHaveLength(2); // tslint:disable-line:no-magic-numbers
+        expect(h4.at(1).props().children).toEqual(ResourceKeys.digitalTwin.steps.secondSuccess);
 
+        const labels = digitalTwinInterfaces.find(Label);
+        expect(labels).toHaveLength(2); // tslint:disable-line:no-magic-numbers
+        expect(labels.at(1).props().children).toEqual([ResourceKeys.deviceInterfaces.columns.source, ': ', ResourceKeys.settings.modelDefinitions.repositoryTypes.local.labelInElectron]);
+
+        const messageBar = digitalTwinInterfaces.find(MessageBar);
+        expect(messageBar).toHaveLength(1);
+        expect(messageBar.first().props().children).toEqual(ResourceKeys.deviceInterfaces.interfaceNotValid);
+    });
+
+    it('shows model id with valid model definition found but has no component from selector', () => {
+        const digitalTwinInterfaces = mountWithLocalization(
+            getComponent({
+                isDigitalTwinLoading: false,
+                modelDefinitionWithSource: {
+                    isModelValid: true,
+                    modelDefinition,
+                    source: REPOSITORY_LOCATION_TYPE.Public,
+                },
+                modelId: 'dtmi:__azureiot:sampleModel;1'
+            }),
+            true,
+            true).find(DigitalTwinInterfaces);
+
+        const h4 = digitalTwinInterfaces.find('h4');
+        expect(h4).toHaveLength(3); // tslint:disable-line:no-magic-numbers
+        expect(h4.at(2).props().children).toEqual(ResourceKeys.digitalTwin.steps.third); // tslint:disable-line:no-magic-numbers
+
+        const labels = digitalTwinInterfaces.find(Label);
         expect(labels).toHaveLength(3); // tslint:disable-line:no-magic-numbers
-        expect(labels.first().props().children).toEqual(ResourceKeys.digitalTwin.dcm);
-        expect(labels.at(1).props().children).toEqual('urn:contoso:com:EnvironmentalSensor:1');
-        expect(labels.at(2).props().children).toEqual('urn:azureiot:ModelDiscovery:DigitalTwin:1'); // tslint:disable-line:no-magic-numbers
+        expect(labels.at(1).props().children).toEqual([ResourceKeys.deviceInterfaces.columns.source, ': ', ResourceKeys.settings.modelDefinitions.repositoryTypes.public.label]);
+
+        expect(digitalTwinInterfaces.find(Announced)).toHaveLength(1);
+        expect(digitalTwinInterfaces.find(Pivot)).toHaveLength(1);
+    });
+
+    it('shows model id with valid model definition found and has components', () => {
+        const digitalTwinInterfaces = mountWithLocalization(
```

Comment:
why do we need `mountWithLocalization` considering there is no localization render prop in the FC?